src/components/auth/CredentialsForm.tsx

```diff
@@ -0,0 +1,117 @@
+import { useFormik } from "formik";
+import { StyleSheet, View } from "react-native";
+import { Button, Input, Link, Text, Title } from "@/components/common";
+import { ArrowRight } from "@/assets/icons";
+import { Colors, Fonts } from "@/styles/theme";
+import { Notification } from ".";
+import { useCreateAccountForm } from "@/services/createAccount";
+import { useNavigation } from "@react-navigation/native";
+import { StackNavigation } from "@/types";
+
+export function CredentialsForm() {
+  const { credentialsSchema, handleNextStep } = useCreateAccountForm();
+  const { navigate } = useNavigation<StackNavigation>();
+  const {
+    handleChange,
+    handleBlur,
+    handleSubmit,
+    values,
+    errors
+  } = useFormik({
+    initialValues: {
+      firstName: "",
+      lastName: "",
+      email: ""
+    },
+    validationSchema: credentialsSchema,
+    onSubmit: handleNextStep,
+    validateOnChange: false,
+    validateOnBlur: false,
+  });
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.textContainer}>
+        <Title>Create your Account</Title>
+        <Text>Enter your details below to start creating your brand new account.</Text>
+      </View>
+      <View style={styles.inputsContainer}>
+        <Input
+          onChangeText={handleChange('firstName')}
+          onBlur={handleBlur('firstName')}
+          placeholder="Enter your first name here"
+          label="Your name"
+          value={values.firstName}
+        />
+        <Input
+          onChangeText={handleChange('lastName')}
+          onBlur={handleBlur('lastName')}
+          placeholder="Enter your last name here"
+          value={values.lastName}
+        />
+        <Input
+          onChangeText={handleChange('email')}
+          onBlur={handleBlur('email')}
+          placeholder="you@email.com.au"
+          value={values.email}
+          label="Email"
+        />
+      </View>
+      {errors.email && (
```

Comment:
instead of manually adding code for error notifications as required its a good pattern to add these to your input component. That way if you ever need a large form with many fields which all need validation and to report their errors you'll be setup well

Reply:
Hum, i didn't understand. Like, I already have an error inside the input component:

<img src="https://github.com/victor-moonward/nbm-forum/assets/158861519/d6098dfa-8bae-4886-9d2d-dec68fe658f3" height="400px" />

Do you mean that would be better to put the entire notification inside the input?